### PR TITLE
Add Support for CCT Last Layer Training with Dim 8-128

### PR DESCRIPTION
**Pull request by runwangdl**
This PR adds support for fine-tuning the last GEMM layer of CCT with embedding dimensions ranging from 8 to 128. The key additions and changes include:

## Added

- Support for SoftmaxCrossEntropyLoss and SoftmaxCrossEntropyLossGrad with tiling.
- Implementation of SGD updates for CCT training.
- Test for one iteration of CCT last-layer training with dimensions from 8 to 128.

## Changed
- Modified the outputs of LayerNorm and SoftmaxCrossEntropyLoss nodes to a single output for better tiling compatibility.
- Added SGD parameter updates to the CCT training graph.

## PR Merge Checklist

1. [x] The PR is rebased on the latest `devel` commit and pointing to `devel`.
2. [x] Your PR reviewed and approved.
3. [x] All checks are passing.
4. [x] The `CHANGELOG.md` file has been updated.
5. [x] If the docker was modified, change back its link after review.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -292,7 +292,7 @@ jobs:
         MLPerf/ImageClassification
         MLPerf/AnomalyDetection
         CCT/CCT_1_16_16_8
-        testTrainCCT/CCT_GEMM_Weight_Bias_1_16_16_8
+        testTrainCCT/CCT_GEMM/CCT_1_16_16_8
```
Comment (Victor-Jung) [inline .github/workflows/CI.yml:295]: Let's think of a better name for the folder. Like `CCT_Classifier_Training`. I'm very open to suggestions on your end also so feel free to propose names :)

File: Deeploy/Targets/Generic/Parsers.py [hunk 2]
```diff
@@ -2429,16 +2423,44 @@ def parseNodeCtxt(self,
...
+class SGDParser(NodeParser):
+
+    def __init__(self):
+        super().__init__()
+
+    def parseNode(self, node: gs.Node) -> bool:
+
+        ret = all([len(node.inputs) == 2, len(node.outputs) == 1])
+
+        return ret
```
Comment (Victor-Jung) [inline Deeploy/Targets/Generic/Parsers.py:2449]: This is a bit too generic no? Isn't there a risk that this parser works for too many nodes? I think the SDG node has several unique attributes like `lr`, etc... One example of checking attributes in the parser is in `iRMSNormParser` of `Generic/Parsers.py`.

File: Deeploy/Targets/PULPOpen/Platform.py [hunk 3]
```diff
@@ -117,10 +116,9 @@
 SoftmaxCrossEntropyLossMapper = NodeMapper(SoftmaxCrossEntropyLossParser(), PULPSoftmaxCrossEntropyTilingReadyBindings)
 SoftmaxCrossEntropyLossGradMapper = NodeMapper(SoftmaxCrossEntropyLossGradParser(),
                                                PULPSoftmaxCrossEntropyGradTilingReadyBindings)
+SGDMapper = NodeMapper(SGDParser(), PULPSGDTilingReadyBindings)
 QuantMapper = NodeMapper(QuantParser(), BasicQuantBindings)
-DequantMapper = NodeMapper(DequantParser(), BasicDequantBindings)
-GEMMDequantMapper = NodeMapper(PULPGEMMParser(), BasicGEMMBindings)
```
Comment (Victor-Jung) [inline Deeploy/Targets/PULPOpen/Platform.py:122]: What happened here? Why merge the two mappers?

File: Deeploy/Targets/PULPOpen/TileConstraints/SGDTileConstraint.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# ----------------------------------------------------------------------
+#
+# File: AddTileConstraint.py
+#
+# Last edited: 21.03.2025
+#
+# Copyright (C) 2023, ETH Zurich and University of Bologna.
+#
+# Author: Moritz Scherer, ETH Zurich
```
Comment (Victor-Jung) [inline Deeploy/Targets/PULPOpen/TileConstraints/SGDTileConstraint.py:9]: Even if Moritz did a great job in Deeploy, you deserve your name here ;)

**Review by Victor-Jung — COMMENTED** [review]
Great job Run! I only have a few minor comments.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (runwangdl) [reply .github/workflows/CI.yml:295]: Solved in [a644fdf](https://github.com/pulp-platform/Deeploy/pull/55/commits/a644fdf47215e709e2052b7c07ad467ae428350a)

File: Deeploy/Targets/Generic/Parsers.py (hunk 2 above)
Comment (runwangdl) [reply Deeploy/Targets/Generic/Parsers.py]: Solved in [a644fdf](https://github.com/pulp-platform/Deeploy/pull/55/commits/a644fdf47215e709e2052b7c07ad467ae428350a) with 'lr'

File: Deeploy/Targets/PULPOpen/Platform.py (hunk 3 above)
Comment (runwangdl) [reply Deeploy/Targets/PULPOpen/Platform.py]: Solved in [643e160](https://github.com/pulp-platform/Deeploy/pull/55/commits/643e16005976609674029eab02941e05a0ea20f1).

File: Deeploy/Targets/PULPOpen/TileConstraints/SGDTileConstraint.py (hunk 4 above)
Comment (runwangdl) [reply Deeploy/Targets/PULPOpen/TileConstraints/SGDTileConstraint.py]: Solved in [a644fdf](https://github.com/pulp-platform/Deeploy/pull/55/commits/a644fdf47215e709e2052b7c07ad467ae428350a)

File: .github/workflows/CI.yml (hunk 1 above)
Comment (Victor-Jung) [reply .github/workflows/CI.yml]: FYI the commit you pointed me at is not the correct one, I found this comment addressed in [this commit](https://github.com/pulp-platform/Deeploy/pull/55/commits/501775d1e34f0d30ebb6f21c4e8d493dc7b4e1f8).

File: Deeploy/Targets/PULPOpen/Platform.py (hunk 3 above)
Comment (Victor-Jung) [reply Deeploy/Targets/PULPOpen/Platform.py]: I'd also like to understand what happened. Was it intentional or not at least?

Comment (runwangdl) [reply Deeploy/Targets/PULPOpen/Platform.py]: This was actually a byproduct of me resolving the conflicts between the two of us, because many of our conflicts couldn’t be automatically merged and had to be resolved manually. Some conflicts were automatically resolved without any notification, and code just got deleted. I think I misunderstood something and ended up changing his code incorrectly.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (runwangdl) [reply .github/workflows/CI.yml]: Sorry Victor, I replied to the comment thinking everything was fine, but later I realized the CI didn’t pass. I haven’t had the chance to update the comment yet.

**Comment by runwangdl** [comment]
My fork CI has passed

**Review by Victor-Jung — APPROVED** [review]
